src/extractor/extractor.cpp

```diff
@@ -336,9 +324,37 @@ std::vector<TurnRestriction> Extractor::ParseOSMData(ScriptingEnvironment &scrip
             }
         });
 
-    // Number of pipeline tokens that yielded the best speedup was about 1.5 * num_cores
-    tbb::parallel_pipeline(tbb::task_scheduler_init::default_num_threads() * 1.5,
-                           buffer_reader & location_cacher & buffer_transform & buffer_storage);
+    // If the user wants coordinates in the way function, we can use
+    // libosmium's NodeLocationsForWays handler to annotate way objects
+    // with cached coordinate data
+    if (scripting_environment.GetProfileProperties().enable_way_coordinates)
+    {
+        using osmium_index_type =
+            osrm::util::SparsePPMemMap<osmium::unsigned_object_id_type, osmium::Location>;
+        using osmium_location_handler_type =
+            osmium::handler::NodeLocationsForWays<osmium_index_type>;
+
+        osmium_index_type location_cache;
+        osmium_location_handler_type location_handler(location_cache);
+
+        // TODO: I don't think that the location cache is thread-safe,
+        //       we probably need to synchronize insertion of nodes
```

Comment:
Just do it in the main thread. The overhead will be much smaller than compared to the synchronization overhead.